### PR TITLE
Testing on sos

### PR DESCRIPTION
**Pull request by thdvbr**
- refactor sos context based on user controller change on the backend 
- refactor controlledtextinput to make it reusable 
- add some unit tests on sos contact list 
- split api logic from soscontext
- delete one of two error components so we can use just one

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,9 +2,9 @@ import 'react-native-gesture-handler';
 import React from 'react';
 import { View, StyleSheet } from 'react-native';
 import { Provider as SosProvider } from 'state/SosContext';
```
Comment (irina-baeva) [inline src/App.js:4]: u can move this to index and then import { AuthProvider , LocationProvider, SosProvider } from 'state/';

File: src/api/index.js [hunk 2]
```diff
@@ -33,12 +33,40 @@ const changePassword = (email, oldPassword, password) => {
...
+};
+const addSosContact = (username, data, token) => {
+  return apiInstance.patch(`/users/${username}/contacts/`, data, {
+    headers: { 'auth-token': token },
+  });
+};
+
+const editSosContact = (username, data, id, token) => {
+  return apiInstance.patch(`/users/${username}/contacts/${id}`, data, {
+    headers: { 'auth-token': token },
```
Comment (irina-baeva) [inline src/api/index.js:56]: can we reuse sending headers so u will not repeat it every time ?

File: src/components/authentication-form/AuthForm.component.js [hunk 3]
```diff
@@ -35,16 +35,19 @@ const AuthForm = ({
...
+            name='username'
+            control={control}
+            placeholder='Username'
+            returnKeyType='next'
+            onSubmitEditing={isEmailFocused}
+            blurOnSubmit={false}
+            inputRef={usernameInputRef}
+            style={styles.input}
+          />
+          <Error name='username' errors={errors} />
```
Comment (irina-baeva) [inline src/components/authentication-form/AuthForm.component.js:49]: what is this error? Aren’t we use from useForm?

File: src/components/form-fields/Error.component.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import styled from 'styled-components/native';
+
+const ErrorText = styled.Text`
+  color: red;
+  bottom: 10;
+`;
+
+const Error = ({ errors, name, style }) => {
```
Comment (irina-baeva) [inline src/components/form-fields/Error.component.js:9]: may be more specific name?

File: src/components/form-fields/Error.component.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import styled from 'styled-components/native';
+
+const ErrorText = styled.Text`
```
Comment (irina-baeva) [inline src/components/form-fields/Error.component.js:4]: better to extract intro error.styles.js

File: src/screens/DeleteAccount/index.js [hunk 6]
```diff
@@ -3,8 +3,8 @@ import { View, Text } from 'react-native';
 import UserInfo from '_components/user-info/';
 import { faAngleLeft } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
-import { StyledView } from '../../styles/shared/StyledView';
 import { AuthContext } from 'state/';
+import { StyledView } from '../../styles/shared/StyledView';
```
Comment (irina-baeva) [inline src/screens/DeleteAccount/index.js:7]: what about alias? it worked for styles i guess

**Review by irina-baeva — COMMENTED** [review]
no failed tests now?

File: src/components/form-fields/Error.component.js (hunk 5 above)
Comment (thdvbr) [reply src/components/form-fields/Error.component.js:4]: Then if i want to customize for different screen i cant use inline styles

**Review by irina-baeva — APPROVED** [review]

**Pull request merged**